lib/Dialect/HW/InstanceGraphBase.cpp

```diff
@@ -51,18 +52,38 @@ InstanceGraphNode *InstanceGraphBase::getOrAddNode(StringAttr name) {
 }
 
 InstanceGraphBase::InstanceGraphBase(Operation *parent) : parent(parent) {
-  parent->walk([&](HWModuleLike module) {
+  SmallVector<std::pair<HWModuleLike, SmallVector<HWInstanceLike>>>
+      moduleToInstances;
+  // First accumulate modules inside the parent op. Use `PreOrder` to avoid
+  // visiting module bodies.
+  parent->walk<mlir::WalkOrder::PreOrder>([&](HWModuleLike module) {
+    moduleToInstances.push_back({module, {}});
+    // Skip.
+    return WalkResult::skip();
+  });
```

Comment:
I think we can use something like `for (auto module : parent->getBody()->getOps<HWModuleLike>()`